### ISSUE TITLE
Error when compiling Bravos Class Hash.

### ISSUE THREAD

**Issue by rodrigo-pino** [lambdaclass/cairo_native#617]
We are starting to execute some transactions accross Mainnet, starting from 0.13.1. We modified the Blockifier to use Native and then hook it up to Juno, in order to replay transactions and compare them with the VM.

When executing transaction [0x54d097cd28a33a15cf90cfe01991a22b167af0167e17f01e0dd30e8d0e6b347](https://voyager.online/tx/0x54d097cd28a33a15cf90cfe01991a22b167af0167e17f01e0dd30e8d0e6b347), since it interacts with Braavos (Account?) with class hash
[0x00816dd0297efc55dc1e7559020a3a825e81ef734b558f03c83325d4da7e6253](https://voyager.online/class/0x00816dd0297efc55dc1e7559020a3a825e81ef734b558f03c83325d4da7e6253) we get the following error from Native, during MLIR generation.

```
thread '<unnamed>' panicked at /Users/rodro/.cargo/git/checkouts/cairo_native-0bad8506281b7281/64d61c4/src/libfuncs.rs:304:51:
index out of bounds: the len is 0 but the index is 0
```

After some digging I think it might be related with Enums when generating an unconditional branch. It's hard for me to delve more into it. Hope this is helpful.

**Comment by varex83** [comment]
Also, some of the TXs that are failing with the similar reason:
`0x651f21d9312659018fbc0fc05b8aacfa26e0faddf3d724064cc251214ccd4f6`
`0x3bb5f95ce60f34258ec423eb92ca17fe5532cf2f7e0ba4ce44b06d0ebb4f5e1`
`0x3e14f39eba2dd853ebf45094bdf2f49066181c4ea66e1fe77fbb4de54edaebb`

**Comment by rodrigo-pino** [comment]
For these transactions:
*  `0x651f21d9312659018fbc0fc05b8aacfa26e0faddf3d724064cc251214ccd4f6` 
 * `0x3bb5f95ce60f34258ec423eb92ca17fe5532cf2f7e0ba4ce44b06d0ebb4f5e1`
 * `0x3e14f39eba2dd853ebf45094bdf2f49066181c4ea66e1fe77fbb4de54edaebb`
 
Native crashes when trying to compile the [Braavos Class Hash](0x00816dd0297efc55dc1e7559020a3a825e81ef734b558f03c83325d4da7e6253)

During this execution, there are other contracts that are successfully compiling. Among them there is the [Argent Class](0x01a736d6ed154502257f02b1ccdf4d9d1089f80811cd6acad48e6b6a9d1f2003)

**Comment by muheebyusufbaba1** [comment]
I'll like to help out on this

**Comment by edg-l** [comment]
We updated cairo-native to use cairo 2.6.3 and LLVM 18 (and Rust 1.78) among other fixes, could you please re-test with this updated version to see if it still doesn't work? (You can use main or this commit 0b3fe0a10d810d5e6d8821f83d635bf914d649ea )

**Comment by rodrigo-pino** [comment]
To throw some light into the issue. At the moment this was unknown to use but when tracing a transaction Juno traces the whole block as well. This means that the mentioned transactions might have worked okay but at some point in the block when executing a Tx that required a Braavos Account it crashed.

We still have to verify execution correctness but this means that there are several transactions running correctly. We started recompiling the [Braavos code ](https://github.com/myBraavos/braavos-account-cairo), removing parts until we find the culprit(s).

**Comment by azteca1998** [comment]
Hello, is the panic location (`libfuncs.rs:304:51`) still accurate? The only reason why it would crash there with that error message, more specifically `the len is 0`, is that we've received a statement without any branches. This can only be the case on return statements, which are handled separately.

If that's the actual issue I think it's been reported earlier in #505. We still don't know why we're receiving those statements without branches. If possible, could you provide us with the smallest program (that you can find) which reproduces the bug? This way we can investigate much faster and potentially delegate the bug report to the correct project if that ends up being the case.

**Comment by varex83** [comment]
Hey there! 

I've prepared small error reproduction of this issue that you can find in this repository https://github.com/varex83/cairo-native-compilation-failed-example-1

Also, duplicating it here:

> When trying to compile with cairo native (with LLVM-18 installed), the following error is thrown:
> 
> ```
> thread 'main' panicked at /cairo_native/src/libfuncs.rs:304:51:
> index out of bounds: the len is 0 but the index is 0
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ``` 
> 
> This error is originates from some_component.cairo, which has no Events, but they're exported in the main contract.cairo
> file.
> 
> To remove the error, the SomeComponent must be removed from the main contract file.
> 
> ---
> 
> This example is a minimal reproduction of the issue that arose when trying to compile
> braavos [account contract](https://github.com/myBraavos/braavos-account-cairo/blob/696405f83c040bb16f102e3cc3274e488a410b4c/src/presets/braavos_account.cairo).
> 
> The following components (that have no events) break the compilation:
> 
> - [SRC5Component](https://github.com/myBraavos/braavos-account-cairo/blob/696405f83c040bb16f102e3cc3274e488a410b4c/src/introspection/src5.cairo)
> - [RateComponent](https://github.com/myBraavos/braavos-account-cairo/blob/e8753ad1eaba0f96abe1f85684c27c3223eaa062/src/dwl/rate_service.cairo)
> - [OutsideExecEvt](https://github.com/myBraavos/braavos-account-cairo/blob/696405f83c040bb16f102e3cc3274e488a410b4c/src/outside_execution/outside_execution.cairo)
>

**Comment by azteca1998** [comment]
Hello, I've compiled the example contract and analized the Sierra. I'll assume you're a bit familiar with the structure of the Sierra programs.

Here's what I've found:
```
type mock_contract::some_component::SomeComponent::Event = Enum<ut@[1112691209375471144620257979429958982113231934115537269132153284534741988113]> [storable: true, drop: true, dup: true, zero_sized: false];
    // Type `mock_contract::some_component::SomeComponent::Event` is an enum without variants. It
    //   cannot be initialized or matched.

type mock_contract::contract::TestingContract::Event = Enum<ut@[726743163660895369335477063681659866259110799100230797474649020819882354857], mock_contract::some_component::SomeComponent::Event, mock_contract::contract::TestingContract::EventA> [storable: true, drop: true, dup: true, zero_sized: false];
    // Type `mock_contract::contract::TestingContract::Event` is an enum with two variants, one of
    //   them being non-initializable. Variant #0 cannot be initialized.

libfunc enum_match<mock_contract::some_component::SomeComponent::Event> = enum_match<mock_contract::some_component::SomeComponent::Event>;
    // This libfunc causes the crash. It attempts to match an unmatchable type.

libfunc enum_init<mock_contract::contract::TestingContract::Event, 1> = enum_init<mock_contract::contract::TestingContract::Event, 1>;
    // This libfunc is OK because it initializes the valid variant of `mock_contract::contract::TestingContract::Event`.
```

And here's the summary:
  - It has an empty enum type (which cannot be initialized because it has no variants).
  - It tries to `enum_match` an instance of that type. This is wrong on many levels:
    - No instance can exist because it cannot be initialized.
    - There's no point in matching since there are no variants.
    - The `enum_match` libfunc has as many branch targets as variants. Since it has no variants there are no target branches, which completely breaks the program's control flow.

**Comment by igaray** [comment]
Issue was closed by CI when #643 was merged, I'll reopen it until we confirm it's progress.

**Comment by azteca1998** [comment]
Hello, I've just tried to run the failing transaction on Juno using the latest native commit (with the fix). It seems the program is not panicking anymore. This is the output (prettified):
```
{
  "jsonrpc": "2.0",
  "error": {
    "code": 63,
    "message": "An unexpected error occurred",
    "data": "execute transaction #50: failed txn 0x03d86ecc867b68b98b5508a2f161bb492fa85ea6a0b00d3dba00fe43078d3598 reason: Transaction validation has failed:\n0: Error in the called contract (contract address: 0x03f4b079641ef45e847b94b7639fcf9dfdde04370be8defb2a95adc1806f86e1, class hash: 0x00816dd0297efc55dc1e7559020a3a825e81ef734b558f03c83325d4da7e6253, selector: 0x0162da33a4585851fe8d3af3c2a9c60b557814e221e0d4f30ff0b2189d9c7775):\nNative execution error: INVALID_SIG\n"
  },
  "id": 0
}
```

I'll close the issue since the compilation error is fixed.
The validation one sounds like an issue in the implementation of the syscall handler. Do you have more information about this?
In any case, feel free to create a new issue to talk about this new one.
